semver.js

```diff
@@ -704,6 +704,42 @@ Comparator.prototype.test = function(version) {
   return cmp(version, this.operator, this.semver, this.loose);
 };
 
+Comparator.prototype.intersects = function(compOrStr, loose, platform) {
+  var rangeTmp;
+  var comp = new Comparator(compOrStr, loose);
```

Comment:
Would it be bad if this only took a Comparator? That would make the API more explicit and delegate more decisions to the user.